addons/binding/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/Connection.java

```diff
@@ -0,0 +1,925 @@
+        request.type = type;
+        request.deviceSerialNumber = device.serialNumber;
+        request.deviceType = device.deviceType;
+        request.createdDate = createdDate;
+        request.alarmTime = alarmTime;
+        request.reminderLabel = label;
+        request.sound = sound;
+        request.originalDate = new SimpleDateFormat("yyyy-MM-dd").format(alarm);
+        request.originalTime = new SimpleDateFormat("HH:mm:ss.SSSS").format(alarm);
+        request.type = type;
+        request.id = "create" + type;
+
+        GsonBuilder gsonBuilder = new GsonBuilder();
+        gsonBuilder.serializeNulls();
+        Gson gson = gsonBuilder.create();
+
+        String data = gson.toJson(request);
+        String response = makeRequestAndReturnString("PUT", alexaServer + "/api/notifications/createReminder", data,
+                true, null);
+        JsonNotificationResponse result = parseJson(response, JsonNotificationResponse.class);
+        return result;
+    }
+
+    public void stopNotification(JsonNotificationResponse notification) throws IOException, URISyntaxException {
+        makeRequestAndReturnString("DELETE", alexaServer + "/api/notifications/" + notification.id, null, true, null);
+    }
+
+    public JsonNotificationResponse getNotificationState(JsonNotificationResponse notification)
+            throws IOException, URISyntaxException {
+        String response = makeRequestAndReturnString("GET", alexaServer + "/api/notifications/" + notification.id, null,
+                true, null);
+        JsonNotificationResponse result = parseJson(response, JsonNotificationResponse.class);
+        return result;
+    }
+
+    public List<JsonMusicProvider> getMusicProviders() {
+        String response;
+        try {
+            Map<String, String> headers = new HashMap<String, String>();
+            headers.put("Routines-Version", "1.1.201102");
+            response = makeRequestAndReturnString("GET",
+                    alexaServer + "/api/behaviors/entities?skillId=amzn1.ask.1p.music", null, true, headers);
+        } catch (IOException | URISyntaxException e) {
+            logger.warn("getMusicProviders fails: {}", e.getMessage());
+            return new ArrayList<>();
+        }
+        if (StringUtils.isEmpty(response)) {
+            return new ArrayList<>();
+        }
+        JsonMusicProvider[] result = parseJson(response, JsonMusicProvider[].class);
+        return Arrays.asList(result);
+    }
+
+    public void playMusicVoiceCommand(Device device, String providerId, String voiceCommand)
+            throws IOException, URISyntaxException {
+        JsonPlaySearchPhraseOperationPayload payload = new JsonPlaySearchPhraseOperationPayload();
+        payload.customerId = device.deviceOwnerCustomerId;
+        payload.locale = "ALEXA_CURRENT_LOCALE";
+        payload.musicProviderId = providerId;
+        payload.searchPhrase = voiceCommand;
+
+        Gson gson = new Gson();
```

Comment:
This is potentially expensive - you should instantiate it only once as a field in that class.